src/commands/type_hierarchy.ts

```diff
@@ -0,0 +1,131 @@
+"use strict";
+
+import * as editors from "../editors";
+import * as vs from "vscode";
+import * as as from "../analysis/analysis_server_types";
+import { Analyzer } from "../analysis/analyzer";
+import { toRange } from "../utils";
+
+export class TypeHierarchyCommand implements vs.Disposable {
+	private context: vs.ExtensionContext;
+	private analyzer: Analyzer;
+	private commands: Array<vs.Disposable> = [];
+
+	constructor(context: vs.ExtensionContext, analyzer: Analyzer) {
+		this.context = context;
+		this.analyzer = analyzer;
+
+		this.commands.push(
+			vs.commands.registerTextEditorCommand("dart.showTypeHierarchy", this.showTypeHierarchy, this)
+		);
+	}
+
+	private showTypeHierarchy(editor: vs.TextEditor, editBuilder: vs.TextEditorEdit) {
+		if (!editors.hasActiveDartEditor()) {
+			vs.window.showWarningMessage("No active Dart editor.");
+			return;
+		}
+
+		let document = editor.document;
+
+		this.analyzer.searchGetTypeHierarchy({
+			file: document.fileName,
+			offset: document.offsetAt(editor.selection.active)
+		}).then(response => {
+			let items = response.hierarchyItems;
+			if (!items) {
+				vs.window.showInformationMessage('Type hierarchy not available.');
+				return;
+			}
+
+			let options = { placeHolder: name(items, 0) };
+
+			// TODO: How / where to show implements?
+			// TODO: How / where to show mixins?
```

Comment:
Don't know; mixins we can probably ignore. W/ implements, people use that quite a bit in Dart to indicate inheritance. One example I was playing around with is `dart:io`'s File and Directory; they both implement FileSystemEntity, but in this case I really think it's more of a subclass arrangement.